code/modules/mob/living/carbon/xenomorph/Abilities.dm

```diff
@@ -993,146 +993,113 @@
 	var/mob/living/carbon/Xenomorph/Queen/X = owner
 	if(!X.check_state())
 		return
-	if(X.observed_xeno)
-		var/mob/living/carbon/Xenomorph/T = X.observed_xeno
-		if(!X.check_plasma(600)) return
+	if(!X.observed_xeno)
+		to_chat(X, "<span class='warning'>You must overwatch the xeno you want to de-evolve.</span>")
+		return
+	
+	var/mob/living/carbon/Xenomorph/T = X.observed_xeno
+	if(!X.check_plasma(600)) return
 
-		if(T.is_ventcrawling)
-			to_chat(X, "<span class='warning'>[T] can't be deevolved here.</span>")
-			return
+	if(T.is_ventcrawling)
+		to_chat(X, "<span class='warning'>[T] can't be deevolved here.</span>")
+		return
 
-		if(!isturf(T.loc))
-			to_chat(X, "<span class='warning'>[T] can't be deevolved here.</span>")
-			return
+	if(!isturf(T.loc))
+		to_chat(X, "<span class='warning'>[T] can't be deevolved here.</span>")
+		return
 
-		if(T.health <= 0)
-			to_chat(X, "<span class='warning'>[T] is too weak to be deevolved.</span>")
-			return
+	if(T.health <= 0)
+		to_chat(X, "<span class='warning'>[T] is too weak to be deevolved.</span>")
+		return
 
-		var/newcaste = ""
-
-		switch(T.caste)
-			if("Hivelord")
-				newcaste = "Drone"
-			if("Carrier")
-				newcaste = "Drone"
-			if("Crusher")
-				newcaste = "Warrior"
-			if("Ravager")
-				newcaste = "Hunter"
-			if("Praetorian")
-				newcaste = "Warrior"
-			if("Boiler")
-				newcaste = "Spitter"
-			if("Spitter")
-				newcaste = "Sentinel"
-			if("Hunter")
-				newcaste = "Runner"
-			if("Warrior")
-				newcaste = "Defender"
-
-		if(!newcaste)
-			to_chat(X, "<span class='xenowarning'>[T] can't be deevolved.</span>")
-			return
+	if(!T.xeno_caste.deevolves_to)
+		to_chat(X, "<span class='xenowarning'>[T] can't be deevolved.</span>")
+		return
+	
+	var/datum/xeno_caste/new_caste = xeno_caste_datums[T.xeno_caste.deevolves_to][1]
 
-		var/confirm = alert(X, "Are you sure you want to deevolve [T] from [T.caste] to [newcaste]?", , "Yes", "No")
-		if(confirm == "No")
-			return
+	var/confirm = alert(X, "Are you sure you want to deevolve [T] from [T.xeno_caste.caste_name] to [new_caste.caste_name]?", , "Yes", "No")
+	if(confirm == "No")
+		return
 
-		var/reason = stripped_input(X, "Provide a reason for deevolving this xenomorph, [T]")
-		if(isnull(reason))
-			to_chat(X, "<span class='xenowarning'>You must provide a reason for deevolving [T].</span>")
-			return
+	var/reason = stripped_input(X, "Provide a reason for deevolving this xenomorph, [T]")
+	if(isnull(reason))
+		to_chat(X, "<span class='xenowarning'>You must provide a reason for deevolving [T].</span>")
+		return
 
-		if(!X.check_state() || !X.check_plasma(600) || X.observed_xeno != T)
-			return
+	if(!X.check_state() || !X.check_plasma(600) || X.observed_xeno != T)
+		return
 
-		if(T.is_ventcrawling)
-			return
+	if(T.is_ventcrawling)
```

Comment:
Inform the queen of these returns, on why she can't devolve the xeno?

Reply:
outside the scope tbh